internal/agent/plugin.go

```diff
@@ -136,7 +136,9 @@ func (self *PluginCommon) gatherDecorations() {
 }
 
 func (self *PluginCommon) decorateEvent(eventData map[string]interface{}) {
-	eventData["timestamp"] = time.Now().Unix()
+	if eventData["timestamp"] == nil {
```

Comment:
this might panic, if `timestamp` key is not present

Reply:
Sorry for approving, i didn't see this may have other implications. But regarding the panic, isn't a "get" on a map a safe operation? If the key does not exist, doesn't it return the "zero" value of the type (which i guess for interface{} it would be nil?

Reply:
@varas this won't panic, the whole point of the nil check is to check if its not present.

Reply:
my bad! you are right map operations are safe whereas map is initialised, sorry for the confusion